test/test-network/systemd-networkd-tests.py

```diff
@@ -8855,6 +8856,45 @@ def check_sysctl_watch(self):
         self.assertNotIn("changed sysctl '/proc/sys/net/ipv6/conf/dummy98/hop_limit'", log)
         self.assertNotIn("changed sysctl '/proc/sys/net/ipv6/conf/dummy98/max_addresses'", log)
 
+class NetworkdRestartTest(unittest.TestCase, Utilities):
+
+    def setUp(self):
+        setup_common()
+
+    def tearDown(self):
+        tear_down_common()
+
+    def test_restart(self):
+        copy_network_unit('12-dummy.netdev', '85-static-ipv6.network')
+        start_networkd()
+        self.wait_online('dummy98:routable')
+
+        # Add an unmanaged interface with an up address
+        call('ip link add unmanaged0 type dummy')
+        call('ip link set unmanaged0 up')
+        call('ip -6 addr add 2001:db8:9999:f101::15/64 dev unmanaged0')
```

Comment:
This should be before wait_online().
You can use `wait_online()` for unmanaged interfaces.

Also, before restarting, maybe better to check 
- the interface is in unmanaged state, and
- the specified address is not removed.